web-marketplace/src/components/organisms/Order/Order.Summary.tsx

```diff
@@ -0,0 +1,189 @@
+import { msg, Trans } from '@lingui/macro';
+import { useLingui } from '@lingui/react';
+
+import { DenomIconWithCurrency } from 'web-components/src/components/DenomIconWithCurrency/DenomIconWithCurrency';
+import { BlockchainIcon } from 'web-components/src/components/icons/BlockchainIcon';
+import CertifiedDocumentIcon from 'web-components/src/components/icons/CertifiedDocumentIcon';
+import CreditsIcon from 'web-components/src/components/icons/CreditsIcon';
+import { PaymentInfoIcon } from 'web-components/src/components/icons/PaymentInfoIcon';
+import { SupCurrencyAndAmount } from 'web-components/src/components/SupCurrencyAndAmount/SupCurrencyAndAmount';
+import QuestionMarkTooltip from 'web-components/src/components/tooltip/QuestionMarkTooltip';
+
+import { Link } from 'components/atoms';
+
+import { OrderSummarySection } from './Order.Summary.Section';
+import { OrderSummaryRow } from './Order.SummaryRow';
+import {
+  BlockchainDetailsData,
+  CreditsData,
+  OrderSummarySectionProps,
+  PaymentInfoData,
+  RetirementInfoData,
+} from './Order.types';
+
+interface OrderSummaryProps {
+  retirementInfo: OrderSummarySectionProps;
+  blockchainDetails: OrderSummarySectionProps;
+  creditsData: OrderSummarySectionProps;
+  paymentInfo: OrderSummarySectionProps;
+}
+
+export const OrderSummary = ({
+  creditsData,
+  retirementInfo,
+  blockchainDetails,
+  paymentInfo,
+}: OrderSummaryProps) => {
+  const { _ } = useLingui();
+  const { purchaseDate, blockchainRecord } =
+    blockchainDetails as BlockchainDetailsData;
+  const { nameOnCard, askDenom, cardLast4 } = paymentInfo as PaymentInfoData;
+  const isCardPayment = nameOnCard && cardLast4 && askDenom === 'usd';
+  const isTradable =
+    (retirementInfo as RetirementInfoData).tradableCredits !== null;
+  const { reason, location, tradableCredits } =
+    retirementInfo as RetirementInfoData;
+  const {
+    credits,
+    price,
+    askDenom: creditAskDenom,
+  } = creditsData as CreditsData;
+
+  const totalPrice = +credits * +price;
+
+  return (
+    <main className="grid grid-cols-1 md:grid-cols-2 gap-30">
+      <OrderSummarySection
+        icon={<CreditsIcon fontSize="medium" className="text-grey-500" />}
+        title={_(msg`Credits`)}
+      >
+        <OrderSummaryRow
+          title={_(msg`Price per credit`)}
+          value={
+            <div className="flex items-center">
+              <SupCurrencyAndAmount
+                price={+price}
+                currencyCode={creditAskDenom}
+                className="mr-10 text-base"
+              />
+              <DenomIconWithCurrency
```

Comment:
please check how I updated it in my PR: https://github.com/regen-network/regen-web/pull/2477/files#diff-6e92c61d1ad4141c61e04825b5cd429147b6a13e6e902f0d95d415002909384c
we need ask**Base**Denom and displayDenom

Reply:
I've updated this to follow what you do in your PR. Please let me know if any further changes are required.